borsh-derive/src/lib.rs

```diff
@@ -59,8 +81,35 @@ pub fn borsh_deserialize(input: TokenStream) -> TokenStream {
     })
 }
 
+fn check_use_discriminant(derive_input: DeriveInput) -> Result<Option<bool>, TokenStream> {
+    for attr in &derive_input.attrs {
+        if attr.path().is_ident("borsh_use_discriminant") {
+            if let Meta::NameValue(value) = attr.meta.clone() {
+                let MetaNameValue {
+                    eq_token: _, value, ..
+                } = value;
+                let value = value.to_token_stream().to_string();
+                return match value.as_str() {
+                    "true" => Ok(Some(true)),
+                    "false" => Ok(Some(false)),
+                    _ => {
+                        return Err(TokenStream::from(
+                            syn::Error::new(
+                                derive_input.ident.span(),
+                                "`use_discriminant` accept only `true` or `false`",
```

Comment:
Please, add a test for this case

Reply:
added in the same file since I am going to do unsplit of those crates